### PR TITLE
add warnings for socket error and unexpected state change

### PR DESCRIPTION
**Pull request by birarda**
- add `qWarning` calls to `udt::Socket` to capture socket state changes and errors

### CODE REVIEW

**Comment by birarda** [comment]
I do not believe any testing is required here so long as the code compiles. No functionality is added - just output to log.

File: libraries/networking/src/udt/Socket.cpp [hunk 1]
```diff
@@ -36,6 +36,11 @@ Socket::Socket(QObject* parent) :
     
     // start our timer for the synchronization time interval
     _synTimer->start(_synInterval);
+
+    // make sure we hear about errors and state chagnes from the underlying socket
```
Comment (sethalves) [inline libraries/networking/src/udt/Socket.cpp]: chagnes

**Comment by birarda** [comment]
Killed that first build, don't be alarmed

**Comment by Nex-Pro** [comment]
@birarda is a killer.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8a3336ae95004e1225b4a2a7fb5a27e974ec84b3 available at: https://gist.github.com/c8047355c6e8df7ca54a36876db37f2d<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7815.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7815.dmg

**Comment by sethalves** [comment]
test this please

**Comment by birarda** [comment]
The PR build succeeded, gist poster failed. Darn.

**Comment by birarda** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8a3336ae95004e1225b4a2a7fb5a27e974ec84b3 available at: https://gist.github.com/c8047355c6e8df7ca54a36876db37f2d<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7815.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7815.dmg

**Comment by ZappoMan** [comment]
Maybe we should do a smoke test? Can we construct a test case that would cause these new log messages to occur? What would cause a socket error or the socket state to change?

I agree it seems extremely unlikely that this change would introduce a problem in the client or server behavior.

If no test is needed, then maybe the dev who approved it should just merge it.

**Pull request merged**